internal/rotate.go

```diff
@@ -93,6 +112,10 @@ func (h *CertRotationHandler) Save() error {
 }
 
 func (h *CertRotationHandler) Rotate() error {
+	var err error
+	defer h.eventSync.NotifyCompleted(err)
+	h.eventSync.NotifyStarted()
```

Comment:
I just want to understand this one:

If any step below fails, are we really completed?

Reply:
its completed with "FAILED". if its re-tried, you get a new set of update events. i tried to make it work like aklite does its update events.

Reply:
OK. My question was really about the _retry_ thing.

IIUC the `handler.eventSync.SetCorrelationId(handler.State.CorrelationId)` line makes sure that any retries will inherit the same correlation ID, right?
That means that on failure we will get smth like this (for the same correlation ID):
```
- some events
- completed with error
- some events
- completed with error
- some events
- completed with success
```

I guess this is exactly how you planned this, right?

Reply:
If your answer is _yes, this is intentional_ then I think we need to document it somewhere.

Also, this means that the case of power outage versus the case of regular error will result into different event chains (power outages will not produce those `completed with error` events).

Reply:
it was bad cherry-picking by me: ea2d92f6e660149d175107397ce4ecdbbd2b90aa

Reply:
it was intentional, but i'm not sure its a good thing. it would be easy to generate a new correlation id each time. that might make this better.

@mike-sul - what do you think?

Reply:
New correlation for each re-try will erase the previous event history, including regular update events, unless I am missing something here.
So, imo it should be the same corID for each re-try with effectively makes `corId` equal to `rotationId`.

Reply:
> New correlation ID for each re-try will erase the previous event history

It won't exactly erase - it will truncate the oldest one >= our history threshold.

So its a matter of - do we want each attempt to be group separately or keep them all together. Right now they are together and doesn't seem terrible to me. There's some weirdness if a retry hits that volodya noted, but that's not the worst thing.

Reply:
>It won't exactly erase - it will truncate the oldest one >= our history threshold.

That's exactly what I meant. Each re-try erases the oldest update event record. If many re-retries happens within short period of time than all previous update events are erased.
So, I opt for the current approach, the same cor-ID for each re-try.

Reply:
34a5dd466e353c037c345ae9b12a5709ef272522

Reply:
This looks good to me.